src/main/java/nextstep/subway/path/domain/PathFinderBuilder.java

```diff
@@ -0,0 +1,51 @@
+package nextstep.subway.path.domain;
+
+import nextstep.subway.line.domain.Section;
+import nextstep.subway.path.application.dto.PathResponse;
+import nextstep.subway.path.application.dto.PathSearchType;
+import nextstep.subway.station.domain.Station;
+import org.jgrapht.graph.WeightedMultigraph;
+
+import java.util.List;
+
+public class PathFinderBuilder {
+    private PathFinder pathFinder;
+    private WeightedMultigraph<Station, CustomWeightedEdge> routeMap;
+    private Station source;
+    private Station target;
+
+    public PathFinderBuilder(PathFinder pathFinder) {
+        this.pathFinder = pathFinder;
+        this.routeMap = new WeightedMultigraph<>(CustomWeightedEdge.class);
+    }
+
+    public PathFinderBuilder addVertex(List<Station> stations) {
+        stations.stream().forEach(this.routeMap::addVertex);
+        return this;
+    }
+
+    public PathFinderBuilder addEdgeWeight(List<Section> sections, PathSearchType type) {
+        sections.stream()
+                .forEach(s -> {
```

Comment:
아래처럼 바로 `forEach()`를 호출해볼 수도 있을 것 같아요.
```suggestion
        sections.forEach(s -> {
```